curated_transformers/util/hf.py

```diff
@@ -87,50 +94,98 @@ def get_model_checkpoint_filepaths(name: str, revision: str) -> List[str]:
     :param revision:
         Model revision.
     :returns:
-        List of absolute paths to the checkpoints.
+        List of absolute paths to the checkpoints
+        and the checkpoint type.
     """
 
-    # Attempt to download a non-sharded checkpoint first.
-    try:
-        model_filename = hf_hub_download(
-            repo_id=name, filename=HF_MODEL_CHECKPOINT, revision=revision
-        )
-    except HTTPError:
-        # We'll get a 404 HTTP error for sharded models.
-        model_filename = None
-
-    if model_filename is not None:
-        return [model_filename]
-
-    try:
-        model_index_filename = hf_hub_download(
-            repo_id=name, filename=HF_MODEL_SHARDED_CHECKPOINT_INDEX, revision=revision
-        )
-    except HTTPError:
-        raise ValueError(
-            f"Couldn't find a valid PyTorch checkpoint for model `{name}` "
-            f"(revision `{revision}`) on HuggingFace Model Hub"
+    def get_checkpoint_paths(
+        checkpoint_type: ModelCheckpointType,
+    ) -> List[str]:
+        checkpoint_type_to_string = {
+            ModelCheckpointType.PYTORCH_STATE_DICT: "PyTorch",
+            ModelCheckpointType.SAFE_TENSORS: "Safetensors",
+        }
```

Comment:
Maybe a method on the enum could return the nice names? Then we don't need an auxiliary mapping.

Reply:
Added.